cli/src/main/kotlin/ls/kafka/backup/s3/BackupBucket.kt

```diff
@@ -87,4 +88,24 @@ class BackupBucket(private val bucket: String, s3Config: S3Config, private val k
             }
         }
     }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    suspend fun getRecordsForOffsets(
+        topic: String,
+        partition: Int?,
+        offsets: Set<Long>,
+    ): Flow<DumpRecord> =
+        s3.backupFiles(bucket, null, Regex(".*")).filter { backupFile -> backupFile.topic == topic }
+            .flatMapConcat { backupFile ->
+                flow {
+                    val allRecords = backupFile.records
+                    val recordsInPartition = if (partition != null) {
+                        allRecords.filter { record -> record.partition == partition }
+                    } else {
+                        allRecords
+                    }
+                    recordsInPartition.filter { record -> record.offset in offsets }.forEach { record -> emit(record) }
```

Comment:
:thinking: I'm not sure that this code works as expected since normally you need to call `.emit()` within a `flow {}` context. Instead, you could say `.asFlow()` after `.filter` which converts the list correctly as flow.